docs/architecture/performance-tests.md

```diff
@@ -0,0 +1,131 @@
+
+#### Example: Install Loggen through syslog-ng
+Refer to your syslog-ng documentation for installation instructions. For example, for Ubuntu:
+
+```bash
+wget -qO - https://ose-repo.syslog-ng.com/apt/syslog-ng-ose-pub.asc | sudo apt-key add -
+
+# Update distribution name
+echo "deb https://ose-repo.syslog-ng.com/apt/ stable ubuntu-noble" | sudo tee -a /etc/apt/sources.list.d/syslog-ng-ose.list
+
+sudo apt-get update
+sudo apt-get install syslog-ng-core
+```
+
+```bash
+loggen -help
+Usage:
+  loggen [OPTION?]  target port
+```
+
+#### Example: Use from Your SC4S Container
+```bash
+sudo podman exec -it SC4S bash
+loggen --help
+Usage:
+  loggen [OPTION*]  target port
+```
+
+## Choose Your Hardware
+Here is a reference example of performance testing using our lab configuration on various types of AWS EC2 machines.
+
+### Tested Configuration
+* Loggen (syslog-ng 3.25.1) - m5zn.3xlarge
+* SC4S(2.30.0) + podman (4.0.2) - m5zn family
+* SC4S_DEST_SPLUNK_HEC_DEFAULT_WORKERS=10 (default)
+* Splunk Cloud Noah 8.2.2203.2 - 3SH + 3IDX
+
+### Command
+```bash
+/opt/syslog-ng/bin/loggen -i --rate=100000 --interval=1800 -P -F --sdata="[test name=\"stress17\"]" -s 800 --active-connections=10 <local_hostmane> <sc4s_external_tcp514_port>
+```
+
+| SC4S instance | root networking                                                                                                     | slirp4netns networking                                                                                                |
+|---------------|---------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
+| m5zn.large    | average rate = 21109.66 msg/sec, count=38023708, time=1801.25, (average) msg size=800, bandwidth=16491.92 kB/sec    | average rate = 20738.39 msg/sec, count=37344765, time=1800.75, (average) msg size=800, bandwidth=16201.87 kB/sec      |
+| m5zn.xlarge   | average rate = 34820.94 msg/sec, count=62687563, time=1800.28, (average) msg size=800, bandwidth=27203.86 kB/sec    | average rate = 35329.28 msg/sec, count=63619825, time=1800.77, (average) msg size=800, bandwidth=27601.00 kB/sec      |
+| m5zn.2xlarge  | average rate = 71929.91 msg/sec, count=129492418, time=1800.26, (average) msg size=800, bandwidth=56195.24 kB/sec   | average rate = 70894.84 msg/sec, count=127630166, time=1800.27, (average) msg size=800, bandwidth=55386.60 kB/sec     |
+| m5zn.2xlarge  | average rate = 85419.09 msg/sec, count=153778825, time=1800.29, (average) msg size=800, bandwidth=66733.66 kB/sec   | average rate = 84733.71 msg/sec, count=152542466, time=1800.26, (average) msg size=800, bandwidth=66198.21 kB/sec     |
+
+## Watch Out for Queues
+While comparing loggen results can be sufficient for A/B testing, it is not enough to accurately estimate the syslog ingestion throughput of the entire system.
+
```

Comment:
## Watch out for gueues
Comparing loggen results can be sufficient for A/B testing, but is not adequate for estimating the syslog ingestion throughput of the entire system.